### ISSUE TITLE
Add support for removing least recently used element

### ISSUE THREAD

**Issue by cswinter** [jeromefroe/lru-rs#8]
Basically I want to use `LruCache` with infinite capacity and unit values just to govern cache eviction policy, and use a criterion other than element count to decide when to/how many elements to remove. This requires a constructor that initializes with `cap: usize::MAX` and `HashMap::default()` and a method like this:

```Rust
    pub fn pop_lru(&mut self) -> Option<(K, V)> {
        let key = KeyRef {
            k: unsafe { &(*(*self.tail).prev).key },
        };
        let mut node = self.map.remove(&key)?;

        let node_ptr: *mut LruEntry<K, V> = &mut *node;
        self.detach(node_ptr);

        // Required because of https://github.com/rust-lang/rust/issues/28536
        let node = *node;
        let LruEntry { key, val, .. } = node;
        Some((key, val))
    }
```

My unsafe Rust is not super strong, so this might be off.

For now I just copied the lib into my project but I'll add some tests and doc comments and make a pull request at some point if you think this is worth merging.

**Comment by jeromefroe** [comment]
Hey! Thanks for the interest in the project! I definitely think this would be worth merging. The code snippet looks good to me, so if you get a chance to make a PR I'd be more than happy to review.

**Comment by jeromefroe** [comment]
Closed by #9. Thanks @cswinter!
